spec/support/shared_examples/persistence_strategy.rb

```diff
@@ -0,0 +1,36 @@
+shared_examples 'a persistence strategy' do
+  shared_context 'with changes' do
+    before do
+      subject.obj << RDF::Statement.new(RDF::Node.new, RDF::DC.title, 'moomin')
+    end
+  end
+
+  describe '#persist!' do
+    xit 'evaluates false with no changes' do
```

Comment:
I think I've remembered why I didn't finish this.  Determining whether a resource changed or not requires either tracking a `@changed` status through an object's lifecycle or grabbing the graph from the persisted source and checking isomorphism.

The latter isn't recommendable, and since we don't currently track `@changed`, this seems out of reach now.

I propose to document that `#persist!` returns `true` or throws an error, for now.  If there are compelling use cases for `@changed`, we can introduce that and update the interface `#persist!` at that time.